packages/gui/src/components/nfts/gallery/NFTGallery.tsx

```diff
@@ -27,6 +27,8 @@ import SelectedActionsDialog from './SelectedActionsDialog';
 
 export const defaultCacheSizeLimit = 1024; /* MB */
 
+export const shownNFTs = new Map();
```

Comment:
We shouldn't export or provide direct access to this object. I would probably use a Context/Provider/hook to wrap this up nicely. You can take a look at LRUsProvider/useNFTMetadataLRU as a possible example. Or I'm sure @seeden would have some excellent advice as well.